### PR TITLE
Rewrites/type hints doc 001

### PR DESCRIPTION
**Pull request by Quix0r**
Changes:
- reformatted some array
- added missing documentation
- added type-hints
- changed double-quotes to single
- and maybe some more

### CODE REVIEW

File: src/Model/Contact/Relation.php [hunk 1]
```diff
@@ -254,12 +261,14 @@ public static function isDiscoverable(string $url, array $contact = [])
 	}
 
 	/**
-	 * @param int $uid   user
+	 * Returns an array of sugguested contacts for given user id
```
Comment (MrPetovan) [inline src/Model/Contact/Relation.php]:
```suggestion
	 * Returns an array of suggested contacts for given user id
```

File: src/Model/Contact/Relation.php [hunk 2]
```diff
@@ -272,20 +281,30 @@ static public function getSuggestions(int $uid, int $start = 0, int $limit = 80)
...
-			['order' => ['last-item' => true], 'limit' => $totallimit]
+			$cid,
+			0,
+			$uid,
+			Contact::FRIEND,
+			Contact::SHARING,
+			Protocol::ACTIVITYPUB,
+			Protocol::DFRN,
+			$diaspora,
+			$ostatus,
```
Comment (MrPetovan) [inline src/Model/Contact/Relation.php]: Please put the parameter values on the same line when their `?` placeholders also are on the same line in the query. This helps with visually mapping values to placeholders.
```suggestion
			$cid,
			0,
			$uid, Contact::FRIEND, Contact::SHARING,
			Protocol::ACTIVITYPUB, Protocol::DFRN, $diaspora, $ostatus,
```

File: src/Model/Post/Link.php [hunk 3]
```diff
@@ -81,59 +89,70 @@ public static function getByLink(int $uri_id, string $url, $size = '')
...
 		}
 		return $url . $id;
 	}
 
-	private static function fetchMimeType(string $url, string $accept = HttpClientAccept::DEFAULT)
+	/**
+	 * Fetches MIME type by URL and Accept: header
+	 *
+	 * @param string $url URL to fetch
+	 * @param string $accept Accept: line
```
Comment (MrPetovan) [inline src/Model/Post/Link.php]:
```suggestion
	 * @param string $accept Comma-separated list of expected response MIME type(s)
```

File: src/Model/Post/Link.php [hunk 4]
```diff
@@ -81,59 +89,70 @@ public static function getByLink(int $uri_id, string $url, $size = '')
...
 		return '';
 	}
 
 	/**
 	 * Add external links and replace them in the body
 	 *
-	 * @param integer $uriid
-	 * @param string $body
+	 * @param integer $uriId URI id
+	 * @param string $body HTML body
```
Comment (MrPetovan) [inline src/Model/Post/Link.php]: You did it again and it's an item body formatted with BBCodes.

File: src/Model/Contact/User.php [hunk 5]
```diff
@@ -82,11 +82,11 @@ public static function insertForContactArray(array $contact)
 	/**
 	 * Apply changes from contact update data to user-contact table
 	 *
-	 * @param array $fields 
-	 * @param array $condition 
-	 * @return void 
-	 * @throws PDOException 
-	 * @throws Exception 
+	 * @param array $fields Fields
+	 * @param array $condition Conditions
```
Comment (MrPetovan) [inline src/Model/Contact/User.php]: You did it again!
```suggestion
	 * @param array $fields
	 * @param array $condition
```

File: src/Model/Post/Link.php [hunk 6]
```diff
@@ -40,36 +40,44 @@ class Link
 	/**
 	 * Check if the link is stored
 	 *
-	 * @param int $uri_id
-	 * @param string $url
-	 * @return bool
+	 * @param int $uriId URI id
+	 * @param string $url URL
```
Comment (MrPetovan) [inline src/Model/Post/Link.php]: And again!
```suggestion
	 * @param int $uriId
	 * @param string $url
```

File: src/Model/Post/Link.php [hunk 7]
```diff
@@ -40,36 +40,44 @@ class Link
...
+		return DBA::exists('post-link', ['uri-id' => $uriId, 'url' => $url]);
 	}
 
-	public static function getByLink(int $uri_id, string $url, $size = '')
+	/**
+	 * Returns URL by URI id and other URL
+	 *
+	 * @param int $uriId URI id
+	 * @param string $url URL
+	 * @param string size Size
```
Comment (MrPetovan) [inline src/Model/Post/Link.php]: Please tell me you didn't do this everywhere.
```suggestion
	 * @param int $uriId
	 * @param string $url
	 * @param string $size
```

File: src/Model/Subscription.php [hunk 8]
```diff
@@ -136,25 +133,25 @@ public static function getPrivateVapidKey(): string
 	/**
 	 * Prepare push notification
 	 *
-	 * @param int $nid
+	 * @param Notification $Notification Notification instance
```
Comment (MrPetovan) [inline src/Model/Subscription.php]: Once more.
```suggestion
	 * @param Notification $Notification
```

File: src/Model/Tag.php [hunk 9]
```diff
@@ -243,28 +243,32 @@ public static function getID(string $name, string $url = '', int $type = null)
 			return $tid;
 		}
 
+		// Also log type and tag id
+		$fields['type'] = $type;
+		$fields['tid'] = $tid;
```
Comment (MrPetovan) [inline src/Model/Tag.php]: At this point `$tid` is empty, not sure it's relevant to include it.

File: src/Model/Tag.php [hunk 10]
```diff
@@ -243,28 +243,32 @@ public static function getID(string $name, string $url = '', int $type = null)
...
 	}
 
 	/**
 	 * Store tag/mention elements
 	 *
-	 * @param integer $uriid URI id
+	 * @param integer $uriId URI id
 	 * @param string $hash Hash
 	 * @param string $name Name
 	 * @param string $url URL
```
Comment (MrPetovan) [inline src/Model/Tag.php]: Twice twice.
```suggestion
	 * @param integer $uriId
	 * @param string $hash
	 * @param string $name
	 * @param string $url
```

File: src/Model/Tag.php [hunk 11]
```diff
@@ -331,52 +335,52 @@ public static function storeFromBody(int $uriid, string $body, string $tags = nu
...
 	}
 
 	/**
 	 * Remove tag/mention
 	 *
-	 * @param integer $uriid URI id
+	 * @param integer $uriId URI id
 	 * @param integer $type Type
 	 * @param string $name Name
 	 * @param string $url URL
```
Comment (MrPetovan) [inline src/Model/Tag.php]: Twice thrice.
```suggestion
	 * @param integer $uriId
	 * @param integer $type
	 * @param string $name
	 * @param string $url
```

File: src/Model/Tag.php [hunk 12]
```diff
@@ -386,35 +390,36 @@ public static function remove(int $uriid, int $type, string $name, string $url =
...
 	 * Remove tag/mention
 	 *
-	 * @param integer $uriid
-	 * @param string $hash
-	 * @param string $name
-	 * @param string $url
+	 * @param integer $uriId URI id
+	 * @param string $hash Hash
+	 * @param string $name Name
+	 * @param string $url URL
```
Comment (MrPetovan) [inline src/Model/Tag.php]: Oh come on.
```suggestion
	 * @param integer $uriId
	 * @param string $hash
	 * @param string $name
	 * @param string $url
```

File: src/Model/Tag.php [hunk 13]
```diff
@@ -386,35 +390,36 @@ public static function remove(int $uriid, int $type, string $name, string $url =
...
-		self::remove($uriid, $type, $name, $url);
+		self::remove($uriId, $type, $name, $url);
 	}
 
 	/**
 	 * Get the type for the given hash
 	 *
-	 * @param string $hash
-	 * @return integer type
+	 * @param string $hash Hash
```
Comment (MrPetovan) [inline src/Model/Tag.php]: This wasn't necessary.
```suggestion
	 * @param string $hash
```

File: src/Model/Tag.php [hunk 14]
```diff
@@ -432,53 +437,53 @@ private static function getTypeForHash(string $hash)
 	/**
 	 * Create implicit mentions for a given post
 	 *
-	 * @param integer $uri_id
-	 * @param integer $parent_uri_id
+	 * @param integer $uriId URI Id
+	 * @param integer $parentUriId Parent URI id
```
Comment (MrPetovan) [inline src/Model/Tag.php]: Yes, I guessed those from the parameter names.
```suggestion
	 * @param integer $uriId
	 * @param integer $parentUriId
```

File: src/Model/Tag.php [hunk 15]
```diff
@@ -432,53 +437,53 @@ private static function getTypeForHash(string $hash)
...
 		DBA::close($tags);
 	}
 
 	/**
 	 * Retrieves the terms from the provided type(s) associated with the provided item ID.
 	 *
-	 * @param int       $item_id
-	 * @param int|array $type
-	 * @return array
+	 * @param int       $uriId URI id
```
Comment (MrPetovan) [inline src/Model/Tag.php]: You didn't have to do this.
```suggestion
	 * @param int   $uriId
```

File: src/Model/Tag.php [hunk 16]
```diff
@@ -432,53 +437,53 @@ private static function getTypeForHash(string $hash)
...
+		$condition = ['uri-id' => $uriId, 'type' => $type];
 		return DBA::selectToArray('tag-view', ['type', 'name', 'url', 'tag-type'], $condition);
 	}
 
 	/**
 	 * Return a string with all tags and mentions
 	 *
-	 * @param integer $uri_id
-	 * @param array   $type
+	 * @param integer $uriId URI id
```
Comment (MrPetovan) [inline src/Model/Tag.php]: Why.
```suggestion
	 * @param integer $uriId
```

File: src/Model/Tag.php [hunk 17]
```diff
@@ -551,30 +561,48 @@ public static function populateFromItem(&$item)
...
-			$search, 0, $uid, Protocol::ACTIVITYPUB, Protocol::DFRN, Protocol::DIASPORA, Protocol::OSTATUS, $uid, 0];
+			$search,
+			0,
+			$uid,
+			Protocol::ACTIVITYPUB,
+			Protocol::DFRN,
+			Protocol::DIASPORA,
+			Protocol::OSTATUS,
+			$uid,
+			0,
```
Comment (MrPetovan) [inline src/Model/Tag.php]:
```suggestion
			$search, 0, $uid,
			Protocol::ACTIVITYPUB, Protocol::DFRN, Protocol::DIASPORA, Protocol::OSTATUS, $uid, 0,
```

File: src/Model/Tag.php [hunk 18]
```diff
@@ -551,30 +561,48 @@ public static function populateFromItem(&$item)
...
```
Comment (MrPetovan) [inline src/Model/Tag.php]:
```suggestion
			$search, 0, $uid,
			Protocol::ACTIVITYPUB, Protocol::DFRN, Protocol::DIASPORA, Protocol::OSTATUS, $uid, 0,
```

File: src/Model/Tag.php [hunk 19]
```diff
@@ -655,7 +686,11 @@ public static function setGlobalTrendingHashtags(int $period, int $limit = 10)
 			FROM `tag-search-view`
 			WHERE `private` = ? AND `uid` = ? AND `uri-id` > ? $block_sql
 			GROUP BY `term` ORDER BY `authors` DESC, `score` DESC LIMIT ?",
-			Item::PUBLIC, 0, $post['uri-id'], $limit);
+			Item::PUBLIC,
+			0,
+			$post['uri-id'],
+			$limit
```
Comment (MrPetovan) [inline src/Model/Tag.php]:
```suggestion
			Item::PUBLIC, 0, $post['uri-id'],
			$limit,
```

File: src/Model/Tag.php [hunk 20]
```diff
@@ -708,7 +743,10 @@ public static function setLocalTrendingHashtags(int $period, int $limit = 10)
 			FROM `tag-search-view`
 			WHERE `private` = ? AND `wall` AND `origin` AND `uri-id` > ? $block_sql
 			GROUP BY `term` ORDER BY `authors` DESC, `score` DESC LIMIT ?",
-			Item::PUBLIC, $post['uri-id'], $limit);
+			Item::PUBLIC,
+			$post['uri-id'],
+			$limit
```
Comment (MrPetovan) [inline src/Model/Tag.php]:
```suggestion
			Item::PUBLIC, $post['uri-id'],
			$limit,
```

File: src/Model/Tag.php [hunk 21]
```diff
@@ -759,13 +797,13 @@ private static function getUIDListByTag(string $tag)
 	/**
 	 * Fetch user who subscribed to the tags of the given item
 	 *
-	 * @param integer $uri_id
+	 * @param integer $uriId URI Id
```
Comment (MrPetovan) [inline src/Model/Tag.php]:
```suggestion
	 * @param integer $uriId
```

File: src/Object/Post.php [hunk 22]
```diff
@@ -739,8 +747,7 @@ protected function getParent()
 	/**
 	 * Set conversation thread
 	 *
-	 * @param Thread $thread
-	 *
+	 * @param Thread $thread Thread to set or NULL
```
Comment (MrPetovan) [inline src/Object/Post.php]: Already inferred from signature.
```suggestion
	 * @param Thread $thread
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Model/Tag.php (hunk 21 above)
Comment (Quix0r) [reply src/Model/Tag.php]: There are 15 such occurrences. I change them all.

File: src/Model/Subscription.php (hunk 8 above)
Comment (Quix0r) [reply src/Model/Subscription.php]: Why are variables sometimes `$Name` and not after common `$camelCase`? Can we just make all even?

File: src/Model/Tag.php (hunk 9 above)
Comment (Quix0r) [reply src/Model/Tag.php]: It was about the variable's type. Should I to a `gettype()` instead?

Comment (MrPetovan) [reply src/Model/Tag.php]: I don't think so, it's the result of `mysql_insert_id()` which has predictable return value types.

File: src/Model/Subscription.php (hunk 8 above)
Comment (MrPetovan) [reply src/Model/Subscription.php]: I told you in another PR, I like to capitalize structured variables in a sea of scalar variables to make them stand out. You can make them all even, I won't stop you, but I feel like we're losing a visual marker of their specificity.

Comment (Quix0r) [reply src/Model/Subscription.php]: Hmm, okay. I personally would name them `$imageInstance` to mark them out. Then I can still have scalar variables like `$uid`. So I better rename them back to capitalized versions to go conform with your coding-convention?

File: src/Model/Tag.php (hunk 9 above)
Comment (Quix0r) [reply src/Model/Tag.php]: Okay, I removed `$tid` from it. Later I also want to add `: void` where `void` is returned.

File: src/Util/Network.php [hunk 23]
```diff
@@ -452,15 +456,15 @@ public static function unparseURL(array $parsed)
 		$scheme    = $get('scheme');
 		$query     = $get('query');
 		$fragment  = $get('fragment');
-		$authority = ($userinfo !== null ? $userinfo."@" : '') .
+		$authority = ($userinfo !== null ? $userinfo.'@' : '') .
```
Comment (MrPetovan) [inline src/Util/Network.php]:
```suggestion
		$authority = ($userinfo !== null ? $userinfo . '@' : '') .
```

File: src/Util/Network.php [hunk 24]
```diff
@@ -452,15 +456,15 @@ public static function unparseURL(array $parsed)
...
 						$get('host') .
-						($port ? ":$port" : '');
+						($port ? ':$port' : '');
```
Comment (MrPetovan) [inline src/Util/Network.php]:
```suggestion
						($port ? ":$port" : '');
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Model/Subscription.php (hunk 8 above)
Comment (MrPetovan) [reply src/Model/Subscription.php]: It's my own personal convention, I don't want to impose it on anyone else (unlike PSR-2). I used it in my own work but I'm not aware about anyone else using it so it should be pretty disparate across the code base.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**